lib/rules/require-expect.js

```diff
@@ -0,0 +1,113 @@
+/**
+ * @fileoverview Require the use of `expect` when using `assert` inside of a
+ * block or when passing `assert` to a function.
+ * @author Mitch Lloyd
+ */
+"use strict";
+
+var utils = require("../utils");
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+module.exports = function (context) {
+    var EXCEPT_SIMPLE_ERROR_MESSAGE = "Should use `assert.expect` when using assertions outside of the top-level test callback",
+        ALWAYS_ERROR_MESSAGE = "Test is missing `assert.expect() call`",
+        currentTest = false;
+
+    function isExpectCall(callee) {
+        return callee.object &&
+               callee.object.type === "Identifier" &&
+               callee.object.name === currentTest.assertName &&
+               callee.property.name === "expect";
+    }
+
+    function isTopLevelExpectCall(callee) {
+        return isExpectCall(callee) && currentTest.blockDepth === 1;
+    }
+
+    function isUsingAssertInNestedBlock(node) {
+        return currentTest.blockDepth > 1 && utils.isAssertion(node.callee, currentTest.assertName);
+    }
+
+    function isPassingAssertAsArgument(node) {
+        for (var i = 0; i < node.arguments.length; i++) {
+            if (node.arguments[i].name === currentTest.assertName) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    function isViolatingExceptSimpleRule(node) {
+        return (isUsingAssertInNestedBlock(node) || isPassingAssertAsArgument(node)) &&
+                !currentTest.isExpectUsed;
```

Comment:
Might want to reverse this condition-- that way, an early call to `assert.expect()` in the user's code will short-circuit this line and avoid calling either of those functions.